jvb/src/main/java/org/jitsi/videobridge/Videobridge.java

```diff
@@ -446,6 +567,47 @@ private void handleColibriRequest(XmppConnection.ColibriRequest request)
         }
     }
 
+    private @NotNull Conference getOrCreateConference(ConferenceModifyIQ conferenceModifyIQ)
+        throws InGracefulShutdownException, XmppStringprepException,
+            ConferenceAlreadyExistsException, ConferenceNotFoundException
+    {
+        String meetingId = conferenceModifyIQ.getMeetingId();
+
+        Conference conference = getConferenceByMeetingId(meetingId);
+
+        if (conferenceModifyIQ.getCreate())
+        {
+            if (conference != null)
+            {
+                logger.warn("Will not create conference, conference already exists for meetingId=" + meetingId);
+                throw new ConferenceAlreadyExistsException();
+            }
+            if (isShutdownInProgress())
+            {
+                logger.warn("Will not create conference in shutdown mode.");
+                throw new InGracefulShutdownException();
+            }
+
+            /* TODO: race condition if something else created a meeting with this meetingId since the lookup above? */
```

Comment:
It looks safe to just encapsulate all of this in `synchronized(conferencesById)`